### DISCUSSION TITLE
Welcome to the discussions for the "Requirements for Self-Organization" paper

### DISCUSSION THREAD

**Discussion by Francesco215** [Francesco215/Language_CA#21] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jkbrooks** [comment]
Wow @Francesco215 I can't believe there is literally no engagement on such an amazing paper.  I really liked your work!

Is there somewhere else people are discussing this?

**Reply by Francesco215** [reply]
Hahahaha thanks a lot for your kind comment!

I'm still new to research so it's normal to start with small steps

There has been some feedback in the tweet [me](https://x.com/FrancescoSacco1/status/1711288381383028943?s=20) and [my lab head](https://x.com/drmichaellevin/status/1711345513247965368?s=20) made when we pubblished the paper, but tweets get old, discussion sections on github last much longer!

**Reply by jkbrooks** [reply]
I got something coming for you bro

**Reply by jkbrooks** [reply]
@Francesco215 https://drive.google.com/drive/u/1/folders/1bTks6f9Rrzdh5YwsfLvl4cazqRC-hwsv

**Reply by Francesco215** [reply]
Wow that's a lot of stuff, I looked at everything, but I'm just going to answer to the stuff you said on the audio, otherwise I'll have too much to write

- At the start of the audio you mentioned that the environment should be dynamic. That is correct, infact when the agent writes his memories in the environment, it changes it. The environment can also have other agents in it, so it can be very dynamic.
- You also mentioned that the information should be stored into a somewhat fractal structure. Probably some kind of structure similar to how files are stored in a computer would be ideal (with folders, sub-folders, and so on until you get to the file you are trying to find). If files in a computer were all in a single folder, they would be much harder to find.
- As for the indexing thing you mentioned: Indexing is one of the most important part of any algorithm of [information retrieval](https://en.wikipedia.org/wiki/Information_retrieval) and there are ways to take a longer document or even a repositoriy and index it to later feed it to an LLM. One interesting example is [Chromadb](https://www.trychroma.com/), however it lacks a fractal way of storing information. Another interesting AI that uses indexing is [Cody](https://about.sourcegraph.com/cody): It basically indexes an entire repo to help you with coding (although it's not that great)
- As for [Neurite](https://github.com/satellitecomponent/Neurite), I have to admit that I never heard of it and it looks very interesting. I'm going to take a deeper look later


Btw, I'm also working on a video explanation for the paper. For now the only part that is close to complete is the part about the ising model section, i would love to have you feedback on that.

Here is the link: https://drive.google.com/file/d/1-SY8O1NWbL96nxdWqVr8eheFRMM2Sc8R/view?usp=sharing

**Comment by jkbrooks** [comment]
In my opinion, your video is great! Although I wouldn’t say I’m an expert
in physics or physics education

Cody looks cool!  We have been studying
https://github.com/BloopAI/bloop but could check out Cody as an alternative.

Yeah, Neurite is one dev right now, I’m hoping to get a little bit of help
for him on some engineering and maybe try testing it out.

**Reply by Francesco215** [reply]
> In my opinion, your video is great! Although I wouldn’t say I’m an expert
in physics or physics education

Thanks!

> Cody looks cool!  We have been studying
https://github.com/BloopAI/bloop but could check out Cody as an alternative.

Nice, I didn't even know it existed, I'll check it out

> Yeah, Neurite is one dev right now, I’m hoping to get a little bit of help
for him on some engineering and maybe try testing it out.

Yeah, and it's not easy to code either

**Comment by jkbrooks** [comment]
By the way, I have to say that it’s really cool to be able to talk to you
about this stuff, I really appreciate cross-disciplinary thinking.  I took
inspiration from your work, and our GitHub chats to get my team and
community to have more node-to-node communications in the hopes that they
will all align, like electrons aligning their spin.

And I’m imagining something similar to be possible with AI agents.  Imagine
that each person has a Dunbar number of 150 but if they manage a group of
AI agents that can connect to more than 150 other agents seamlessly an
ongoing real time communication, maybe what I am doing is increasing how
many human nodes in a social graph can align with other humans

Then there is a parallel for getting information nodes in an information
network to connect with increasing number of other information nodes, which
I believe is the point of your LLM comparison.

**Reply by Francesco215** [reply]
>  like electrons aligning their spin.

Hhahahah exactly

Btw, how cool would it be if all paper had a discussion section? 
If there is something you don't understand, you just go and go to the discussion section where people are talking about it.

> Imagine that each person has a Dunbar number of 150 but if they manage a group of AI agents that can connect to more than 150 other agents seamlessly an ongoing real time communication, maybe what I am doing is increasing how many human nodes in a social graph can align with other humans.

Yep, basically everyone will have a team of AI spokesperson/secretaries to handle communications for us. Nowadays only rich people can do that

**Comment by jkbrooks** [comment]
How is your research going @Francesco215 ?  I'm about to cite you in our group's plan to deploy a large set of agents that use an environment for long-term memory and figured I should check in.  

In what new directions are you taking your focus?

**Reply by Francesco215** [reply]
Hi man! good to see you!

Now I'm working for a startup doing research, hopefully we'll have something cool to show soon. Until then it's a secret

>  I'm about to cite you in our group's plan to deploy a large set of agents that use an environment for long-term memory and figured I should check in.

Uhh that seems cool, do you have something cool show off?

**Reply by jkbrooks** [reply]
We're going to drop a blog post in the next 1-5 days or so, I'll share a link.

I'm excited to hear about your startup when you're able to share

**Reply by jkbrooks** [reply]
@Francesco215 Here it is!

I'd love to hear any feedback https://siftransmitter.substack.com/p/an-introduction-to-sif

**Reply by Francesco215** [reply]
ohh seems interesting, I'll take a look later!
